### ISSUE TITLE
iOS keyboard covers bottom right Floating Action Button (FAB)

### ISSUE THREAD

**Issue by AndrWeisR** [ionic-team/ionic#13801]
**Ionic version:**  (check one with "x")
(For Ionic 1.x issues, please use https://github.com/ionic-team/ionic-v1)
[ ] **2.x**
[ x] **3.x**
[ ] **4.x**

**I'm submitting a ...**  (check one with "x")
[ x] bug report
[ ] feature request

Please do not submit support requests or "How to" questions here. Instead, please use one of these channels: https://forum.ionicframework.com/ or http://ionicworldwide.herokuapp.com/

**Current behavior:**

I have a form with an input field, and a floating action button labelled "Save" which sits at the bottom of the page. On Android, when the focus is in the input field and the keyboard pops up, the FAB moves up to sit above the keyboard.
On iOS, however, the FAB does not move up and disappears behind the keyboard.

Note I am testing on an iPhone 6 using the Ionic View app.

**Expected behavior:**

On iOS, the FAB should move up to sit above the keyboard as it does on Android.

Before focus on input field:
![screen1](https://user-images.githubusercontent.com/34931916/34810823-7eea0442-f6ed-11e7-8453-33c6a78bd660.png)

When keyboard pops up on iOS, it obscures the FAB:
![screen2](https://user-images.githubusercontent.com/34931916/34810822-7eb4e262-f6ed-11e7-8024-aae66a5ee91f.png)

On Android, the FAB remains visible above the keyboard:
![screenshot_1515651045](https://user-images.githubusercontent.com/34931916/34810992-651b62a8-f6ee-11e7-8df8-852fbf080cbd.png)

**Steps to reproduce:**

**Related code:**

My template:
```
<ion-header>
  <ion-navbar>
    <ion-title>Edit Claim Details</ion-title>
  </ion-navbar>
</ion-header>

<ion-content padding>
    <ion-fab right bottom>
        <button ion-fab color="primary" (click)="save()">Save</button>
    </ion-fab>
    
    <form novalidate [formGroup]="form">
        <ion-item>
            <ion-label stacked>Injured Person's Name</ion-label>
            <ion-input formControlName="injuredPersonName"></ion-input>
          </ion-item>
    </form>
</ion-content>

```

**Other information:**

**Ionic info:** (run `ionic info` from a terminal/cmd prompt and paste output below):

```
cli packages:
    @ionic/cli-utils  : 1.19.0
    ionic (Ionic CLI) : 3.19.0

global packages:

    cordova (Cordova CLI) : 7.1.0

local packages:

    @ionic/app-scripts : 3.1.0
    Cordova Platforms  : android 6.3.0 ios 4.5.4
    Ionic Framework    : ionic-angular 3.9.2

System:

    Android SDK Tools : 26.1.1
    Node              : v8.9.3
    npm               : 5.6.0
    OS                : Windows 8.1

Environment Variables:

    ANDROID_HOME : C:\Android

Misc:

    backend : pro
```

**Comment by diebarral** [comment]
I had the same issue today and I found a workaround. 

I started by defining in my controller a boolean variable that indicates if the keyboard is open or not. I defined it in `false` by default and in the constructor I subscribed to the `onKeyboardShow` and `onKeyboardHide` events (from the Ionic Keyboard), so I can change my variable accordingly. 

Then I created a function that returns me a class name if the platform is iOS and the keyboard is open, so: 

```javascript
fabButtonClass() {
  return this.platform.is('ios') && this.keyboardOpen ? 'ios-fab-button' : '';
}

```

And the HTML code for the fab button would be: 
```
<ion-fab right bottom [ngClass]="fabButtonClass()">
    <button ion-fab color="primary" (click)="save()">Save</button>
</ion-fab>
```

And in the stylesheet: 
` .ios-fab-button { top: 45%; } `

I hope this works.
